bin/cache_repo

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+set -e
+
+AWS_BUCKET=$1
+
+DATE=$(date +"%Y-%m-%d")
+
+mkdir -p /tmp/$DATE
+
+# Download the repo
+git clone --mirror $BUILDKITE_REPO "/tmp/$DATE/$BUILDKITE_PIPELINE_SLUG.git"
+
+# Create the tarball
+tar -C "/tmp/$DATE" -cvf $DATE.git.tar $BUILDKITE_PIPELINE_SLUG.git
+
+# Copy the file to S3
+aws s3 cp $DATE.git.tar s3://$AWS_BUCKET/$BUILDKITE_ORGANIZATION_SLUG/$BUILDKITE_PIPELINE_SLUG/$DATE.git.tar
```

Comment:
I _think_ this would work without the file name for the S3 destination, too:

```
aws s3 cp $TAR_NAME $S3_BUCKET_PATH
```

I'm saying this because I upload the assets for [mokacoding](https://mokacoding.com) with a command like

```
for f in $(ls ~/Desktop/2021-06-07*.jpg); do aws s3 cp $f s3://mokacoding/ --acl public-read; done
```

But, it doesn't hurt to leave it. 🤷‍♂️ 👍

Reply:
This was another case of "better safe than sorry" 😅